src/main/java/net/sf/jabref/exporter/SaveActions.java

```diff
@@ -72,4 +148,36 @@ private Formatter getFormatterFromString(String formatterName) {
         return new IdentityFormatter();
     }
 
+    public static String getMetaDataString(List<FieldFormatterCleanup> actionList) {
+        //first, group all formatters by the field for which they apply
+        HashMap<String, List<String>> groupedByField = new HashMap<>();
+        for (FieldFormatterCleanup cleanup : actionList) {
+            String key = cleanup.getField();
+
+            // add new list into the hashmap if needed
+            if (!groupedByField.containsKey(key)) {
+                groupedByField.put(key, new ArrayList<>());
+            }
+
+            //add the formatter to the map if it is not already there
+            List<String> formattersForKey = groupedByField.get(key);
+            if (!formattersForKey.contains(cleanup.getFormatter().getKey())) {
+                formattersForKey.add(cleanup.getFormatter().getKey());
+            }
+        }
+
+        // convert the contents of the hashmap into the correct serialization
+        StringBuilder result = new StringBuilder();
+        for (String fieldKey : groupedByField.keySet()) {
```

Comment:
Coverity will complain about inefficient iterator (and I will change it...), so you might want to do (something like):

``` java
for (Entry<String, List<String>> entry : groupedByField.entrySet()) {
result.append(entry.getValue().stream().collect(Collectors.joining(",", entry.getKey() + "[" , "]")));  // Or use StringJoiner, not sure if stream() is really needed here.
}
```